packages/web3-hooks/base/src/useContext.tsx

```diff
@@ -8,29 +8,33 @@ import { useActualProviderType } from './useProviderType.js'
 
 interface Web3Context<T extends NetworkPluginID> {
     account?: string
+    networkPluginId?: T
     chainId?: Web3Helper.Definition[T]['ChainId']
+    setChainId?: (chainId: Web3Helper.Definition[T]['ChainId']) => void
     networkType?: Web3Helper.Definition[T]['NetworkType']
     providerType?: Web3Helper.Definition[T]['ProviderType']
 }
 
 const PluginIDContext = createContext(NetworkPluginID.PLUGIN_EVM)
 
-const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(EMPTY_OBJECT)
+// Avoid use this context directly
+export const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(EMPTY_OBJECT)
 
 const PluginsWeb3Context = createContext<Record<NetworkPluginID, Web3Helper.Web3State<NetworkPluginID>>>(null!)
 
 export function PluginIDContextProvider({ value, children }: React.ProviderProps<NetworkPluginID>) {
+    console.log('------------------useContext PluginIDContextProvider')
+    console.log(value)
     return <PluginIDContext.Provider value={value}>{children}</PluginIDContext.Provider>
 }
 
 export function PluginWeb3ContextProvider<T extends NetworkPluginID>({
-    pluginID,
     value,
     children,
-}: {
-    pluginID: T
-} & React.ProviderProps<Web3Context<T>>) {
-    return <PluginWeb3Context.Provider value={value} children={children} />
+}: React.ProviderProps<Omit<Web3Context<T>, 'setChainId'>>) {
+    const [chainId, setChainId] = useState<Web3Helper.Definition[T]['ChainId'] | undefined>(value.chainId)
```

Comment:
We as well need to set `account` `providerType`, and `networkType`.

Reply:
I added the account / providerType setter.  And I think should not care the networkType in Web3ContextProvider.